### ISSUE TITLE
Fallout 2 (38410)

### ISSUE THREAD

**Issue by dreamer** [ValveSoftware/Proton#1774]
# Compatibility Report
- Name of the game with compatibility issues: Fallout 2
- Steam AppID of the game: 38410

## System Information
- GPU: Intel® HD Graphics 3000
- Driver/LLVM version: 3.0 Mesa 18.0.5
- Kernel version: 4.18.13
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/dreamer/4f3773020ded88d4612e41de4dd67b7e
- Proton version: 3.16-3 Beta

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

## Symptoms

The same as for Fallout, described in #1503:

- Game launcher does not work out of the box. Using protontricks to install/repair dotnet40 may fix it (worked for me in 3.7, does not seem to work in 3.16)
- Input issue - cursor stops responding to input for several several seconds, every few seconds (worked in 3.7, regression in 3.16)

## Workarounds

- You can create a symbolic link to run fallout2HR.exe instead of Fallout2Launcher.exe

**Comment by Pharmasolin** [comment]
Hi @dreamer, meanwhile this is not merged, how do you create a symbolic link to able run the game in steam and not in system installed wine?
Thanks!

**Comment by dreamer** [comment]
Played for few minutes using 3.16-4 and regression with mouse input seems gone.

**Comment by BloodyIron** [comment]
* Ubuntu 18.10
* GTX 960 v418.56
* Proton 4.2.1

Fallout 2 crashes on initial load. No GUI screen comes up after selecting first option.

**Comment by dreamer** [comment]
As far as I can tell, release 4.2-3 fixes the launcher, game itself works like native. Steam overlay does not work (behaviour is the same as in Fallout 1, as I mentioned [here](https://github.com/ValveSoftware/Proton/issues/1503#issuecomment-484690836)), but it's not a problem.

@kisak-valve Ready to be Whitelisted maybe?

**Comment by BloodyIron** [comment]
I'll have to try that out then! :D

**Comment by kisak-valve** [comment]
Hello, is anyone seeing an issue that would block this game from being whitelisted?

**Comment by kisak-valve** [comment]
Fallout 2 (38410) ignores ddraw.ini and possibly f2_res.ini

Issue transferred from https://github.com/ValveSoftware/Proton/issues/2675.
@chloekek posted on 2019-05-11T23:41:09:

# Compatibility Report
- Name of the game with compatibility issues: Fallout 2
- Steam AppID of the game: 38410

## System Information
- GPU: Intel Corporation Device 591b (rev 04)
- Driver/LLVM version: Intel Open Source Technology Center Mesa DRI Intel(R) HD Graphics 630 (Kaby Lake GT2) x86/MMX/SSE2
- Kernel version: 4.19.34
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/chloekek/c03df6ba16e834a7420e8823dc2e9da7
- Proton version: 4.2

## I confirm:
- [ ] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[steam-38410.log](https://github.com/ValveSoftware/Proton/files/3169539/steam-38410.log)

The logs hint at ddraw.dll being taken from C:\windows\\system32\\ddraw.dll rather than the patched version in ~/.local/share/Steam/steamapps/common/Fallout 2. I tried playing around with Wine DLL overrides in the Proton settings but I have too little knowledge about this.

## Symptoms 

Changing settings in ddraw.ini and f2_res.ini (both found in ~/.local/share/Steam/steamapps/common/Fallout 2) don't seem to have any effect. I tried the following options in ddraw.ini and they did not have any effect:

 - SkipOpeningMovies=1 (this one is easy to test with)
 - ProcessorIdle=30 (but CPU usage was still over 150%)

Other than these mods not being applied the game seems to run normally and is playable, however constantly using over 150% CPU according to htop.

## Reproduction

When starting the game from Steam it asks whether you want to play Fallout 2 or the classic version. Both versions exhibit the issue described above.

**Comment by madewokherd** [comment]
Wine's default behavior is usually to prefer builtin dlls. An override of ddraw=n should change that, but I don't know exactly how to set that in Proton.

**Comment by kisak-valve** [comment]
@madewokherd, WINEDLLOVERRIDES is mentioned on other issue reports, perhaps adding `WINEDLLOVERRIDES="ddraw=n" %command%` to the game's launch options will have an effect.

**Comment by dreamer** [comment]
Tried:

    PROTON_LOG=1 WINEDLLOVERRIDES="ddraw=n" %command%

Launcher starts, but game itself does not - screen is black and then Gnome complaints about unresponsive app. In the log, `ddraw.dll` provided with the game was picked:

    150607.933:002e:002f:trace:module:load_dll looking for L"ddraw.dll" in L"Z:\\home\\dreamer_\\.local\\share\\Steam\\steamapps\\common\\Fallout 2;C:\\Program Files (x86)\\Steam;C:\\windows\\system32;C:\\windows\\system;C:\\windows;.;C:\\windows\\system32;C:\\windows;C:\\windows\\system32\\wbem"
    150607.933:002e:002f:trace:module:load_dll Found L"Z:\\home\\dreamer_\\.local\\share\\Steam\\steamapps\\common\\Fallout 2\\DDRAW.dll" for L"ddraw.dll" at 0x330000, count=3

Then, the whole log is filled with 

    …:fixme:d3d:wined3d_texture_add_dirty_region Ignoring dirty_region (0, 0, 0)-(256, 256, 1).

[steam-38410.log](https://github.com/ValveSoftware/Proton/files/3174741/steam-38410.log)

[edit] After forcing DX9 in ddraw.ini (tried Mode=4 and Mode=5). The game now starts, but performance is atrocious - it's unplayable.

**Comment by ritalat** [comment]
I only have the old pre-bethesda GOG version of the game but I've done a few playthroughs under Wine with it. I'm not sure how different the later Steam release is. I doubt they changed much.

In order to get it working nicely with Wine you have to get a more recent version of the included sfall ddraw.dll from [https://github.com/phobos2077/sfall](https://github.com/phobos2077/sfall). Any version that has a "SingleCore" option in ddraw.ini should be good.

After replacing the included old sfall ddraw with that, set mode=4 (or 5) like @dreamer mentioned and also set SingleCore=0. This should fix the abysmal performance and choppy mouse movement.

**Comment by dreamer** [comment]
@ritalat Fallout 2 already works really well using Proton with default settings. The only missing thing is Steam Overlay support.

That being said, some people might miss functionality provided by sfall in ddraw.ini, so I tested the latest release of this mod (4.1.7). With default settings screen is black, as before. With settings you suggested: SingleCore=0 and mode=4 - game works well again (performance is great), but there's still no Steam Overlay support.

**Comment by kisak-valve** [comment]
For transparency, the request to switch this game over to a whitelist request was blocked by the Steam Overlay issue. I'm not sure if the .ini config discussion would be a blocker because it's not really clear if these directly affect the out-of-the-box experience vs advanced tuning.

**Comment by dreamer** [comment]
Turns out lack of Steam Overlay is not a bug - publisher **disabled** this feature by setting [DisableOverlay=Yes](https://steamdb.info/app/38410/info/). I did not test F2 from a different source, but overlay works ok in GOG version of the Fallout 1 added as a shortcut.

**Comment by kisak-valve** [comment]
This game was found in the 2019-07-31 whitelist update.

If you encounter an issue with this game, please search the open issues for 38410, and report any issues encountered with this game in separate issue reports, one issue per issue report.

**Comment by shoober420** [comment]
# Whitelist Request
- Name of the game to be whitelisted: Fallout 2
- Steam AppID of the game: https://steamdb.info/app/38410/

## System Information
- GPU: GTX 760
- Driver/LLVM version: NVIDIA 440.36
- Distro version: Arch
- https://github.com/shoober420/linux-scripts/blob/main/home/shoober420/sysinfonvidia
- Proton version: 4.11-8

Works out of the box. Some tweaks are needed though, like the forementioned ```WINEDLLOVERRIDES="ddraw=n"``` to get sfall to load. Besides that, it runs fantastic. Be sure to update sfall to the latest version before playing. I also use "PROTON_USE_D9VK=1". You still must set "SingleCore=0" as mentioned prior to fix laggy mouse.
